time-split/control.lua

```diff
@@ -1 +1,179 @@
+                tick_to_timestamp(tick))
+        end
+        table.insert(global.players[player_index]["reference_checkpoints"],
+            { name = name, tick = tick })
+    end
+end
+
+script.on_event(defines.events.on_player_created, function(event)
+    populate_globals()
+
+    local player = game.get_player(player_index)
+    local screen_element = player.gui.screen
+    local main_frame = screen_element.add {
+        type = "frame",
+        name = "timesplit_mainframe",
+        caption = "Time Split Table",
+    }
+
+    main_frame.style.size = { 350, 350 }
+    main_frame.add {
+        type = "scroll-pane",
+        name = "content_frame",
+        direction = "vertical"
+    }
+end)
+
+script.on_event(defines.events.on_player_crafted_item, function(event)
+    local item_name = event.item_stack.name
+    local player_global = global.players[player_index]
+    local reference_checkpoints = player_global["reference_checkpoints"]
+
+    for _, ref_entry in pairs(reference_checkpoints) do
+        if (ref_entry.name == item_name) and not contains(player_global["current_checkpoints"], item_name) then
+            table.insert(player_global["current_checkpoints"], { name = ref_entry.name, tick = game.ticks_played })
+        end
+    end
+end)
+
+script.on_event(defines.events.on_research_finished, function(event)
+    local technology_name = event.technology.name
+    local player_global = global.players[player_index]
+    local reference_checkpoints = player_global["reference_checkpoints"]
+
+    for _, ref_entry in pairs(reference_checkpoints) do
+        if ref_entry.name == technology_name then
+            table.insert(player_global["current_checkpoints"], { name = ref_entry.name, tick = game.ticks_played })
+        end
+    end
+end)
+
+script.on_event(defines.events.on_rocket_launched, function(event)
+    local player_global = global.players[player_index]
+
+    if player_global.rocket_launched == false then
+        table.insert(player_global["current_checkpoints"], { name = "rocket-launch", tick = game.ticks_played })
+        player_global["rocket_launched"] = false
+    end
+end)
+
+
+function create_table(frame, ticks)
+    local table = frame.add { type = "table", name = "table", column_count = 4, draw_vertical_lines = true }
+    local player_global = global.players[player_index]
+    local reference_checkpoints = player_global["reference_checkpoints"]
+    local current_checkpoints = player_global["current_checkpoints"]
+    local timesplit_set = false
+    local player = game.get_player(player_index)
+
+
+    for _, ref_entry in pairs(reference_checkpoints) do
+        table.add { type = "label", caption = space_padded(ref_entry.name) }
+        local diff = ticks - ref_entry.tick
+
+        local current_diff_label = table.add { type = "label", caption = "-" }
+        if not timesplit_set then
+            local sign = "+"
+            if diff < 0 then
+                sign = "-"
+                diff = diff * -1
```

Comment:
Would be nice if the value of sign could be reflected in a color (i.e. "-" = red and "+" = green)

Reply:
https://github.com/lars-petter-hauge/time-split/issues/2